### DISCUSSION TITLE
Uderstanding/documenting stats

### DISCUSSION THREAD

**Discussion by ssardina** [idrave/FOND-ASP#27] [Q&A] [answered]
Hi @idrave 

I re-run the prob02 problematic example. I used 32GB ram with 10 threads, and gave the correct SAT answer:

![image](https://user-images.githubusercontent.com/2368384/132338884-9df52934-c891-4638-a2db-90e431803ca2.png)

As you can see it took 40' in total but I want to understand the meaning of each time reported.

What I don't understand is what is `Time`: 1308.946

I think I got the others. In fact the wall time was  pre-processing time + solve time. And I think CPU time is when you add up all the threads. 

So what I don't get is what `Time` reports.

Finally, `Max Memory` says 0. But we know it uses more than 8G for sure... what is that measuring then? I want to get all this right so I can document it in the README.

Thanks!

I am glad this complex example is wroking. Clearly the number of threads used is very important eh!

**Comment by idrave** [comment, marked as answer]
Hello @ssardina ,

Here `Time` is the time that Clingo took to solve the problem, as given in stdout by Clingo itself. Actually, if you notice the 41' approximates better to pre-processing+Time.
Solve time and CPU Time are also in the Clingo output, the solve time is `Time` excluding Clingo's command preprocessing (not the one we do on `Pre-processing Time`) and grounding, so the amount of time the clingo command took in solving exclusively.
As for CPU Time, I understand it is what you say, the total CPU time over threads for solving.

The `Max Memory` was bugged, I must have introduced a bug in one of the latest updates, but I found it and fixed it earlier today. We will have to run it again to know how much memory it used :-). The memory metric of the output is in MBs.

**Reply by ssardina** [reply]
great! Great explanation. I will explain this in the README, thanks!

Great also we killed another bug (the memory report)
